### PR TITLE
[opt](inverted index) Optimization of the initialization process in topn

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Proposed changes

1. reduce the performance cost of initialization

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -2387,14 +2388,16 @@ Status SegmentIterator::_next_batch_internal(vectorized::Block* block) {
...
-    RETURN_IF_ERROR(_init_current_block(block, _current_return_columns));
-    _converted_column_ids.assign(_schema->columns().size(), 0);
 
-    _current_batch_rows_read = 0;
     uint32_t nrows_read_limit = _opts.block_row_max;
     if (_can_opt_topn_reads()) {
         nrows_read_limit = std::min(static_cast<uint32_t>(_opts.topn_limit), nrows_read_limit);
     }
+
+    RETURN_IF_ERROR(_init_current_block(block, _current_return_columns));
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:2397]: nrows_read_limit is not passed to _init_current_block()

**Review by xiaokang — APPROVED** [review]
LGTM

**Comment by xiaokang** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Pull request merged**